src/test/java/core/basesyntax/service/dataservice/PurchaseDataServiceTest.java

```diff
@@ -0,0 +1,54 @@
+package core.basesyntax.service.dataservice;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.dao.FruitDao;
+import core.basesyntax.dao.FruitDaoImpl;
+import core.basesyntax.database.Storage;
+import core.basesyntax.model.FruitTransaction;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+class PurchaseDataServiceTest {
+    private static DataService dataService;
+    private static FruitTransaction fruitTransaction;
+
+    @BeforeAll
+    static void beforeAll() {
+        FruitDao fruitDao = new FruitDaoImpl();
+        dataService = new PurchaseDataService(fruitDao);
+    }
+
+    @AfterEach
+    void tearDown() {
+        Storage.fruitsStorage.clear();
+    }
+
+    @Test
+    void processTransaction_incorrectQuantity_notOk() {
+        fruitTransaction = new FruitTransaction("p","apple",-1);
+        assertThrows(RuntimeException.class,
+                () -> dataService.processTransaction(fruitTransaction));
+    }
+
+    @Test
+    void processTransaction_notEnoughQuantity_notOk() {
+        fruitTransaction = new FruitTransaction("p","apple",3);
+        assertThrows(RuntimeException.class,
+                () -> dataService.processTransaction(fruitTransaction));
+        Storage.fruitsStorage.put("apple",2);
+        fruitTransaction = new FruitTransaction("p","apple",3);
```

Comment:
we need one more assert after these lines, isn't it?

Reply:
ofcourse :)